io/js/src/main/scala/fs2/io/net/tls/TLSSocketPlatform.scala

```diff
@@ -38,16 +38,29 @@ private[tls] trait TLSSocketCompanionPlatform { self: TLSSocket.type =>
 
   private[tls] def forAsync[F[_]](
       socket: Socket[F],
-      upgrade: fs2.io.Duplex => facade.tls.TLSSocket
+      upgrade: fs2.io.Duplex => facade.tls.TLSSocket,
+      clientMode: Boolean
   )(implicit F: Async[F]): Resource[F, TLSSocket[F]] =
     for {
-      duplexOut <- mkDuplex(socket.reads)
-      (duplex, out) = duplexOut
-      _ <- out.through(socket.writes).compile.drain.background
-      tlsSockReadable <- suspendReadableAndRead(
-        destroyIfNotEnded = false,
-        destroyIfCanceled = false
-      )(upgrade(duplex))
+      tlsSockReadable <- socket match {
+        case Socket.AsyncSocket(sock, _) if clientMode =>
```

Comment:
In theory the same logic should work for server sockets as well, but it results in a hang. I think this is connected to the pause or readable state of the socket vs duplex but I can't figure out how to unpause the socket at the right moment to allow things to flow.